### PR TITLE
Ai debugging

### PR DESCRIPTION
**Pull request by vincentjguo**
I think the basics of the AI is done

### CODE REVIEW

**Comment by potatoeggy** [comment]
the dreaded merge conflict

**Comment by vincentjguo** [comment]
I know

**Comment by vincentjguo** [comment]
how to resolve?

File: src/AI.java [hunk 1]
```diff
@@ -1,66 +1,185 @@
-import java.util.LinkedList;
-import java.util.Queue;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.PriorityQueue;
 
-class coordinate {
-    int x, y, color, rowSize;
+class coordinate implements Comparable<coordinate> {
```
Comment (potatoeggy) [inline src/AI.java:5]: Reminder to reformat to follow Java naming conventions later

File: src/Board.java [hunk 2]
```diff
@@ -39,6 +40,8 @@ public boolean checkWin(int x, int y, int player) {
                 else luF = true;
             }
         }
+        /*System.out.println(cL + " " + cU + " " + cRD + " " + cRU);
```
Comment (potatoeggy) [inline src/Board.java:43]: Comments like this should not be in master or prod

File: src/Board.java [hunk 3]
```diff
@@ -3,9 +3,10 @@
     int[][] board = new int[H][W];
 
     public boolean checkWin(int x, int y, int player) {
-        int cL = 0, cU = 0, cRU = 0, cRD = 0;
+        if (board[x][y] != player) return false;
```
Comment (potatoeggy) [inline src/Board.java:6]: m e a n i n g f u l v a r i a b l e n a m e s

File: src/test.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+import java.io.InputStreamReader;
+import java.util.Arrays;
+import java.util.StringTokenizer;
+
+public class test {
+    public static void main(String[] args) throws IOException {
+        int board[][] = new int[6][7];
+        Board b = new Board();
+        board[5][0] = board[5][2] = board[5][3] = 1;
+        /*board[5][0] = 2;
```
Comment (potatoeggy) [inline src/test.java:12]: Remove comments

File: src/test.java [hunk 5]
```diff
@@ -0,0 +1,58 @@
+import java.io.BufferedReader;
```
Comment (potatoeggy) [inline src/test.java:1]: Remove file

**Comment by potatoeggy** [comment]
I'll handle the conflicts

File: src/AI.java (hunk 1 above)
Comment (vincentjguo) [reply src/AI.java:5]: Ok I'll reopen that issue again.

**Comment by vincentjguo** [comment]
Wait i already fixed it, it was just the MainWindow file

**Pull request merged**